[team11][iOS] - 홈 화면 구현

wnsxor1993 opened the pull request:
# [team11][iOS] - 홈 화면 구현
안녕하세요 두두! team11의 푸코와 Sol입니다.
저희는 이번 주에 새로운 프로젝트를 시작하면서 SPM을 처음 사용하다보니 문제가 생각보다 많이 발생해 생각보다 진행 사항이 적었습니다.
그래서 이번 PR은 View 위주로 제일 첫 화면을 구성한 내용만 포함이 되어있습니다.

## 💡 Issue
1. Storyboard 없이 code로만 View 구현
2. UISearchBar를 사용하여 searchBar 구현
    * searchBar click 시, 숙소 찾기 화면으로 넘어가는 기능 구현
    * scroll 시, searchBar가 사라지도록 기능 구현
3. UICollectionViewCompositionalLayout을 이용해 Home화면 구현

## 📝 Tasks

## Task1

#### 🔹 작업
- `UISearchBar`를 사용하여 searchBar 구현

#### 🔹 고민과 해결
- NavigationBar에 searchBar를 구현하는 부분에 있어 `UISearchBar`를 활용하는 방향으로 접근하였으나, 차후 숙소 찾기 화면에서 사용될 searchBar는 UISearchController를 활용해야할 듯 함. 재사용 문제로 인해 현재 구현된 방향은 Controller를 사용하는 방향으로 차후 수정해보고자 함.
- 스크롤 시에 searchBar가 사라지는 것이 더욱 깔끔해 보일 듯 하여 `hidesBarsOnSwipe` 메서드를 통해 해당 기능을 구현.


### Task2

#### 🔹 작업
- `UICollectionViewCompositionalLayout`을 이용해 각각의 Section에 다른 형태의 Cell이 들어갈 수 있도록하여, Home 화면 전체를 하나의 UICollectionView로 구현했습니다.
#### 🔹 고민과 해결 
- CompositionalLayout 객체를 반환하는 메소드의 길이가 너무 길어져 ViewController에 꼭 해당 메소드가 있어야 하는지에 대한 의문이 생겼습니다.
  UICollectionView와 같은 VC에 들어갈 뷰 관련 부분을 따로 `HomeView` 라는 클래스로 빼서, CompositionalLayout 객체를 반환하는 메소드, collectionView의 레이아웃을 설정하는 메소드를 VC로부터 분리했습니다.


#### 🔹 실행 화면
![SS 2022-05-25 PM 02 22 49](https://user-images.githubusercontent.com/92504186/170186043-03ce3bbd-93df-4712-918e-d56c454bb007.gif)

wnsxor1993 committed:
feat: 프로젝트 재생성

Hansolkkim committed:
docs: .gitignore 파일 생성

Hansolkkim committed:
feat: Alamofire Framework Dependency 추가

wnsxor1993 committed:
feat: cocoapod 설정

Hansolkkim committed:
fix: 빌드 안되는 문제 해결

wnsxor1993 committed:
feat: TabBarController 구현 및 Storyboard 제거 #2

wnsxor1993 committed:
feat: TabBarItem 연결할 ViewController들 구현 #2

wnsxor1993 committed:
feat: TabBar sceneDelegate 연결 및 bar 배경색 변경 #2

Hansolkkim committed:
Merge pull request #3 from wnsxor1993/feat/TabBarController

[iOS] TabBarController 생성

wnsxor1993 committed:
feat: SearchVC navigationBar에 searchBar 추가 #4

Hansolkkim committed:
feat: SearchVC의 히어로이미지에 해당하는 Cell 구현 #5

wnsxor1993 committed:
feat: searchBar 클릭 시, 다음 화면으로 push 구현 #4

Hansolkkim committed:
feat: SearchVC의 "가까운 여행지" 해당하는 Cell 구현 #5

Hansolkkim committed:
feat: SearchVC의 "살아보는거야" 해당하는 Cell 구현 #5

wnsxor1993 committed:
refactor: VC 이동 modal로 변경 #4

wnsxor1993 committed:
refactor: modal 이동 시, 신규 navigationController 생성 구현 #4

Hansolkkim committed:
feat: SearchVC의 CollectionView가 사용할 DataSource 구현 #5

Hansolkkim committed:
chore: 임시적으로 Cell에서 사용할 이미지 Assets에 추가 #5

Hansolkkim committed:
feat: SearchVC의 CollectionView HeaderView로 사용될 뷰 구현 #5

Hansolkkim committed:
feat: SearchVC의 CollectionView 구현 및 Cell들의 속성 Figma에 맞춰 조정 #5

Hansolkkim committed:
Merge pull request #6 from wnsxor1993/feat/SearchVC_searchBar

[iOS] SearchVC - searchBar 구현

wnsxor1993 committed:
Merge branch 'iOS-develop' into feat/SearchVC-CollectionView

wnsxor1993 committed:
Merge pull request #7 from wnsxor1993/feat/SearchVC-CollectionView

[iOS] searchVC의 collectionView 구현

wnsxor1993 committed:
refactor: VC 이동 push로 변경 및 tabBar hidden 추가

wnsxor1993 committed:
refactor: SearchVC scrolling시, searchBar hidden 구현

Hansolkkim committed:
refactor: dequeueReusableCell로 이용되는 Cell, HeaderView들의 identifier 프로퍼티 추가

Hansolkkim committed:
refactor: UIColor extension으로 자주 사용되는 색상 정의 및 사용

Hansolkkim committed:
refactor: Cell들의 이름을 좀 더 명확하게 수정

Hansolkkim committed:
refactor: SearchVC에서 View 관련 내용들을 HomeView로 분리

Hansolkkim committed:
chore: 필요없는 위치(Section 0)에 있는 HeaderView 제거

Hansolkkim committed:
refactor: UIView에서 여러 subView들을 추가하는 메소드 extension으로 추가

wnsxor1993 committed:
Merge pull request #9 from wnsxor1993/refactor/SearchVC-CollectionView

[iOS] refactor: SearchVC의 View 관련 수정

Hansolkkim committed:
Merge pull request #8 from wnsxor1993/refactor/SearchVC_searchBar

[iOS] refactor: SearchVC vc 이동 수정

wnsxor1993 committed:
refactor: test

wnsxor1993 committed:
refactor: private extension 분리 #10

wnsxor1993 committed:
refactor: CollectionHeaderView font 관련 메서드 수정 #10

wnsxor1993 committed:
refactor: SearchBar View 분리 #10

wnsxor1993 committed:
refactor: 파일 그룹화 #10

Hansolkkim committed:
Merge pull request #11 from wnsxor1993/refactor/MainView

[iOS] Refactor: Home 화면 전체 리팩토링

Hansolkkim committed:
style: SwiftLint 경고창 없애기 위한 코드 수정 #10

Hansolkkim committed:
Merge branch 'refactor/MainView' into iOS-develop

wnsxor1993 committed:
feat: BrowseVC에 searchBar 구현 #12

wnsxor1993 committed:
feat: BrowseVC collectionView 초기 구현 #12

wnsxor1993 committed:
refactor: SearchVC, BrowseVC searchBar 분리 #12

Hansolkkim committed:
feat: 숙소 찾기 화면의 틀이 될 화면 구현 #13

wnsxor1993 committed:
feat: BrowseVC collectionView layout 설정 #12

Hansolkkim committed:
chore: Pod 관련 수정 #13

캘린더 등록을 위해 HorizonCalendar pod 추가
SwiftLint의 적용에서 벗어나는 예외 설정이 가능한 yml파일 추가

Hansolkkim committed:
feat: SelectCalendarView에 캘린더 추가 및 선택된 날에 맞게 DayRange를 표시해주는 기능 구현 #13

Hansolkkim committed:
chore: 파일 폴더 구분 및 날짜 선택 후 나타날 BudgetView 파일 추가 #13

Hansolkkim committed:
feat: Delegate를 이용해 날짜가 선택되면 FindAccomodationView로 알려주는 흐름 구현 #13

mienne commented on iOS/Airbnb_iOS/Airbnb_iOS/Extension/UIColor+.swift:
```diff
@@ -0,0 +1,19 @@
+import UIKit
+
+extension UIColor {
+    static let gray1 = UIColor(red: 51/255, green: 51/255, blue: 51/255, alpha: 1.0)
```
최소 버전이 13 이상이라면 Color Asset에서 정의하고 코드로 확장해서 사용하는 방법도 있습니다!

[🙂 참고: ColorSet 세팅하여 편리하게 사용하는 방법 (RGB, custom color)](https://ios-development.tistory.com/686)

Hansolkkim replied:
ColorSet에 대해 처음 배웠습니다! 
해당 방법이 extension 했을 때 더 깔끔한 느낌이 있네요. 감사합니다ㅎㅎ
말씀 주신 방법대로 수정해서 한 번 사용해보겠습니다!

mienne commented on iOS/Airbnb_iOS/Airbnb_iOS/Extension/UiView+.swift:
```diff
@@ -0,0 +1,16 @@
+import UIKit
+
+extension UIView {
+    func addSubViews(_ subViews: [UIView]) {
```
개인적인 취향이지만, `func addSubViews(_ views: UIView...)` 선호합니다👀

Hansolkkim replied:
기본 문법 공부할때 봤을 법한 Variadic Parameters 인데 처음 보는 듯한 느낌이 있네요...
안그래도 제가 작성한 메소드 호출할 때 불필요하게 배열로 view들을 묶어서 보낸다는 느낌이 있었는데, 리뷰어님이 선호하시는 방식이 조금 더 보기에도 좋을 것 같습니다! 감사합니다!

mienne commented on iOS/Airbnb_iOS/Airbnb_iOS/View/SearchView/HomeView.swift:
```diff
@@ -0,0 +1,136 @@
+        ])
+    }
+
+    func getCollectionViewLayout() -> UICollectionViewCompositionalLayout {
```
CollectionViewLayout 생성하는 로직을 알아야 하는 주체가 ViewController에서 HomeView로 옮겨졌을 뿐, HomeView에서도 CollectionViewLayout 만드는 로직을 알 필요 없다고 생각합니다. Layout 만드는 객체를 만들어서 위임해야 역할이 제대로 나눠진거라고 생각합니다!

```swift
func getCollectionViewLayout() -> UICollectionViewCompositionalLayout {
    UICollectionViewCompositionalLayout { (section, _) -> NSCollectionLayoutSection? in
        return HomeCollectionViewLayout(section: section).create()
    }
}

struct HomeCollectionViewLayout {
    
    private let section: Int
    
    init(section: Int) {
        self.section = section
    }
    
    func create() -> NSCollectionLayoutSection? {
        switch section {
        case 0:
            return HomeZeroCollectionLayoutSection().create()
        case 1:
            return HomeOneCollectionLayoutSection().create()
        default:
            return HomeDefaultCollectionLayoutSection().create()
        }
    }
}

struct HomeDefaultCollectionLayoutSection {
    
    func create() -> NSCollectionLayoutSection? {
        let itemFractionalWidthFraction = 0.8
        let itemInset: CGFloat = 2.5
        
        let itemSize = NSCollectionLayoutSize(
            widthDimension: .fractionalWidth(itemFractionalWidthFraction),
            heightDimension: .fractionalHeight(1))
        let item = NSCollectionLayoutItem(layoutSize: itemSize)
        item.contentInsets = NSDirectionalEdgeInsets(top: 28, leading: 16, bottom: 24, trailing: 0)
        
        let group = NSCollectionLayoutGroup.vertical(layoutSize: .init(widthDimension: .estimated(335),
                                                                       heightDimension: .estimated(420)),
                                                     subitem: item, count: 1)
        
        let section = NSCollectionLayoutSection(group: group)
        section.contentInsets = NSDirectionalEdgeInsets(top: itemInset, leading: itemInset,
                                                        bottom: itemInset, trailing: itemInset)
        let headerSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1.0),
                                                heightDimension: .absolute(50.0))
        let header = NSCollectionLayoutBoundarySupplementaryItem(layoutSize: headerSize, elementKind: UICollectionView.elementKindSectionHeader, alignment: .top)
        section.boundarySupplementaryItems = [header]
        section.orthogonalScrollingBehavior = .continuous
        
        return section
    }
}

struct HomeZeroCollectionLayoutSection {
    
    func create() -> NSCollectionLayoutSection? {
        let itemFractionalWidthFraction = 1.0
        let groupFractionalHeightFraction = 1.0 / 2.0
        let itemInset: CGFloat = 0
        
        let itemSize = NSCollectionLayoutSize(
            widthDimension: .fractionalWidth(itemFractionalWidthFraction),
            heightDimension: .fractionalHeight(1))
        let item = NSCollectionLayoutItem(layoutSize: itemSize)
        item.contentInsets = NSDirectionalEdgeInsets(top: itemInset, leading: itemInset,
                                                     bottom: 40, trailing: itemInset)
        
        let groupSize = NSCollectionLayoutSize(
            widthDimension: .fractionalWidth(1.0),
            heightDimension: .fractionalHeight(groupFractionalHeightFraction))
        let group = NSCollectionLayoutGroup.horizontal(layoutSize: groupSize, subitems: [item])
        
        let section = NSCollectionLayoutSection(group: group)
        section.contentInsets = NSDirectionalEdgeInsets(top: itemInset, leading: itemInset,
                                                        bottom: itemInset, trailing: itemInset)
        
        return section
    }
}


struct HomeOneCollectionLayoutSection {
    
    func create() -> NSCollectionLayoutSection? {
        let itemFractionalWidthFraction = 0.8
        let itemInset: CGFloat = 2.5
        
        let itemSize = NSCollectionLayoutSize(
            widthDimension: .fractionalWidth(itemFractionalWidthFraction),
            heightDimension: .fractionalHeight(1))
        let item = NSCollectionLayoutItem(layoutSize: itemSize)
        item.contentInsets = NSDirectionalEdgeInsets(top: itemInset, leading: 16,
                                                     bottom: 24, trailing: itemInset)
        
        let group = NSCollectionLayoutGroup.vertical(layoutSize: .init(widthDimension: .fractionalWidth(0.75),
                                                                       heightDimension: .estimated(200)),
                                                     subitem: item, count: 2)
        
        let section = NSCollectionLayoutSection(group: group)
        section.contentInsets = NSDirectionalEdgeInsets(top: itemInset, leading: itemInset,
                                                        bottom: itemInset, trailing: itemInset)
        let headerSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1.0),
                                                heightDimension: .absolute(50.0))
        let header = NSCollectionLayoutBoundarySupplementaryItem(layoutSize: headerSize, elementKind: UICollectionView.elementKindSectionHeader, alignment: .top)
        section.boundarySupplementaryItems = [header]
        section.orthogonalScrollingBehavior = .continuous
        
        return section
    }
}
```

Hansolkkim replied:
HomeView가 Layout을 만드는 방법에 대해서 알 필요가 없다는 말씀을 보고 코드에 대한 고민이 조금 부족했지 않나 생각이 들었습니다..
직접 분리하는 코드도 짜주셔서 정말 감사합니다! 예시가 없었으면 어떻게 나누는지 한참을 생각했을 것 같습니다..
해당 방법처럼 Layout 만드는 객체를 HomeView로 부터 분리해보도록 하겠습니다!

mienne reviewed (commented):
안녕하세요. 리뷰 담당하게 된 두두라고 합니다🙂 

1차적으로 고민, 해결 위주로 보고 리뷰 진행하였습니다.
26일 저녁 10시 이후로 좀 더 상세하게 보도록 할게요!

궁금한 부분은 언제든지 코멘트 남겨주세요👏

wnsxor1993 committed:
refactor: HomeViewCollectionLayout 로직 분리

Hansolkkim committed:
refactor: UIColor Extension 방법 수정

기존에 사용했던 RGB 값을 하드코딩해서 extension하는 방법에서 Assets의 ColorSet 등록을 이용해 extension하는 방법으로 수정

Hansolkkim committed:
refactor: UIView extension 메소드인 addSubViews(_:) 메소드 수정

addSubView(_:) 메소드의 매개변수를 [UIView] 타입에서 가변 매개변수로 UIView타입으로 받는 방향으로 수정

wnsxor1993 committed:
refactor: file grouping 진행

Hansolkkim committed:
chore: SwiftLint를 위한 코드 수정

Hansolkkim committed:
Merge pull request #14 from wnsxor1993/refactor/firstReview

[iOS] Refactor: 첫번째 리뷰 사항 수정

wnsxor1993 commented:
안녕하세요^^
바쁘신 와중에도 신경써서 리뷰 남겨주셔서 감사합니다😁😁
1차로 남겨주신 리뷰 내용을 반영하여 수정 코드 다시 푸시하였습니다.
확인 부탁드리겠습니다!ㅎㅎ

wnsxor1993 committed:
feat: navigation 이동 시, 바로 커서 이동 및 키보드 출력 구현

wnsxor1993 committed:
feat: collectionView click 시, keyboard dismiss 구현

wnsxor1993 committed:
feat: mapkit searching 기능 구현

mienne commented on iOS/Airbnb_iOS/Airbnb_iOS/View/CollectionHeaderView.swift:
```diff
@@ -0,0 +1,59 @@
+        let label = UILabel()
+        label.textColor = .black
+        label.textAlignment = .left
+        label.numberOfLines = 0
```
1. CollectionView의 헤더 사이즈가 50으로 고정되어 있어서 label 높이가 50보다 커지면 대응하지 못할거 같아요.
2. label 이라는 변수명이 범용적이기 때문에 구체적인 역할이 드러나도록 이름으로 변경하는 것은 어떨까요?

Hansolkkim replied:
> 1. CollectionView의 헤더 사이즈가 50으로 고정되어 있어서 label 높이가 50보다 커지면 대응하지 못할거 같아요.

해당 부분에 대해 고민해보고, 아래의 2가지 방법을 생각했습니다. 
> 1. label의 fontSize와 HeaderView의 높이를 고려해서 label text의 numberOfLines 결정하기
> 2. HeaderView의 height를 동적으로 설정하기

여기서 2번 방법이 적절하다고 판단해(HeaderView의 높이는 layout에서 설정해주기 나름이므로, 50이라고 단정지어 생각할 수 없다고 생각했습니다.) 해당 방법으로 수정해보려고 합니다!

> 2. label 이라는 변수명이 범용적이기 때문에 구체적인 역할이 드러나도록 이름으로 변경하는 것은 어떨까요?

HeaderView에 들어가는 View가 UILabel 하나라, 큰 생각없이 범용적인 이름으로 결정했던 것 같습니다. 해당 View의 이름을 ContentLabel으로 수정하겠습니다!

mienne replied:
솔이 생각한 방향 중에서도 타이틀 같은 영역은 동적 처리보다 고정 높이로 처리하는 케이스도 많아서 추후, 디자이너, 기획자 있는 프로젝트 진행하게 된다면 커뮤니케이션하면서 조율하면 됩니다🙂

mienne commented on iOS/Airbnb_iOS/Airbnb_iOS/View/CollectionHeaderView.swift:
```diff
@@ -0,0 +1,59 @@
+        NSLayoutConstraint.activate([
+            label.centerYAnchor.constraint(equalTo: centerYAnchor),
+            label.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 16),
+            label.trailingAnchor.constraint(equalTo: trailingAnchor, constant: 16)
```
```suggestion
            // trailing 조건이 -16이 되어야 좌우 여백이 동일하게 처리됩니다🙂
            label.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -16)
```

AutoLayout 어렵지 않지만, 생소한 부분이라고 생각해요. 그래서 사실 한 번 공부할 때 빡세게 하면 편한 부분이 아주 많습니다. 학습 기간동안 아래 영상 보면서 정리하면 좋지 않을까 싶어요. 권고사항이니 시간 있으시면 보는 것을 추천드려요!🙂

* [인프런의 iOS AutoLayout을 활용한 실전 UI구성 전략 - 카카오톡 같은 고급 UI 만들기](https://www.inflearn.com/course/autolayout-ui_ios#curriculum) 
* WWDC Video가 나온지 시간이 조금 되었지만, AutoLayout 영상

Hansolkkim replied:
안그래도 오토레이아웃에 관해 부족함이 있어 조금 더 자료를 찾아보면서 공부하려고 계획했었는데 추천 감사합니다!
해당 부분 말씀처럼 수정하고, 오토레이아웃 빡세게 공부해보겠습니다! 감사합니다!

mienne commented on iOS/Airbnb_iOS/Airbnb_iOS/ViewController/TabBarItemVC/SearchViewController.swift:
```diff
@@ -0,0 +1,42 @@
+class SearchViewController: UIViewController {
+
+    let searchBar = SearchBarView()
+    let nextVC = BrowseViewController()
```
nextVC의 변수명은 무엇을 의미하나요?

wnsxor1993 replied:
다음으로 push되는 ViewController라는 의미에서 작성한 변수명인데, 의미가 불명확한 느낌이 있어 다음 작업 시에 수정하고자 합니다.

mienne commented on iOS/Airbnb_iOS/Airbnb_iOS/ViewController/TabBarItemVC/SearchViewController.swift:
```diff
@@ -0,0 +1,42 @@
+extension SearchViewController: UISearchBarDelegate {
+    func searchBarShouldBeginEditing(_ searchBar: UISearchBar) -> Bool {
+        self.nextVC.hidesBottomBarWhenPushed = true
+        self.navigationController?.pushViewController(nextVC, animated: true)
```
1. 검색어 입력하려고 할 때, 다른 화면에서 검색 입력을 유도하려고 한다면 UISearchBar 상속받는 것보다 UIView를 상속받아 커스텀하게 구성하는 것이 자유도가 높아 좀 더 낫지 않을까 생각합니다.
2. 그래도 홈 화면 내에서 바로 검색하려는게 의도라면 UISearchBar 사용할 때 사용자에게 키보드 UX 어떻게 제공하는지 다른 앱에서도 한 번 사용해보시길 추천드립니다🙂

mienne replied:
2번째 부분은 이미 많은 상용 서비스에서 사용자가 입력하는 UI가 있을 때 키보드 노출/미노출도 필수로 같이 고려해야 하는 부분이기 때문에 말씀 드려요!

wnsxor1993 replied:
말씀하신 부분은 한 번 찬찬히 확인하면서 어떻게 수정하면 좋을 지 고민해보도록 하겠습니다ㅎㅎ

mienne reviewed (approved):
수고하셨습니다!

Hansolkkim committed:
feat: FindAccomodationVC에서 사용할 extension, usecase, entity 파일 추가

Hansolkkim committed:
feat: View-Controller-UseCase 간 로직 구현

기존의 로직에서 MVC 패턴에 더 걸맞는 로직으로 체크인/체크아웃 날짜를 결정하는 로직 변경
기존에는 SelectCalendarView가 체크인/체크아웃 날짜를 결정하는 로직을 가지고 있어 View의 역할을 벗어난다 판단하여, 해당 로직을 Controller와 UseCase가 가지도록 수정

wnsxor1993 committed:
refactor: 변수명 변경

wnsxor1993 committed:
refactor: 위치 정보 주소로 출력

Hansolkkim committed:
feat: Calendar를 현재 날짜로부터 1년 후까지 보여주도록 설정하는 코드 추가

Hansolkkim committed:
design:  Calendar에서 선택한 두 날짜 모두 동그란 마크가 생기도록 설정 및 날짜 간 간격 조절

wnsxor1993 committed:
Merge branch 'iOS-develop' into feat/firstBrowseView

Hansolkkim committed:
Merge pull request #15 from wnsxor1993/feat/firstBrowseView

[iOS] 숙소 찾기 화면 구현 (인기 스팟 -> 지역 검색)

wnsxor1993 committed:
refactor: PR merge 시, 누락된 파일 재추가

Hansolkkim committed:
Merge branch 'iOS-develop' into feat/secondBrowseView

wnsxor1993 committed:
Merge pull request #16 from wnsxor1993/feat/secondBrowseView

[iOS] 체크인/ 체크아웃 날짜 결정하는 캘린더 뷰 구현

Hansolkkim committed:
refactor: PR merge 시, 누락된 파일 재추가

Hansolkkim committed:
fix: merge 과정 중 반영하지 않았던 내용 추가 반영

Hansolkkim committed:
chore: SwiftLint 준수를 위한 코드 정리

wnsxor1993 commented:
다음 PR을 위해 부득이하게 먼저 merge를 하고자 합니다ㅠㅠ
말씀주신 사항들은 계속 신경 쓰면서 코드에 반영하도록 하겠습니다^^

mienne commented on iOS/Airbnb_iOS/Airbnb_iOS/DateData.swift:
```diff
@@ -0,0 +1,33 @@
+
+import Foundation
+
+struct DateData {
```
DateData보다 Date 표기방법을 변환하는 객체 역할 같은데, DateUtil 이나 DateConvertor 이름이 어떨까요?

mienne commented on iOS/Airbnb_iOS/Airbnb_iOS/View/FindAccomodationView/SelectCalendarView.swift:
```diff
@@ -35,7 +36,17 @@ final class SelectCalendarView: UIView {
             }
-            self.selectedDay = day
+            if self.selectedDay1 != nil &&
+                self.selectedDay2 != nil {
```
어떤 조건인지 드러나면 좋을거 같아요. `let isSelectedBothDays = selectedDay1 != nil && selectedDay2 != nil`
